index.bs

```diff
@@ -734,7 +753,271 @@ Each {{WebRequestEvent}} has:
+  |request|, run the following steps:
+
+  1. Let |environment settings object| be |request|'s [=request/client=].
+
+  1. Let |crossOriginIsolatedCapability| be |environment settings object|'s
+      [=environment settings object/cross-origin isolated capability=], or
+      false if |environment settings object| is null.
+
+  1. Let |details| be a new {{WebRequestEventDetails}} with the following
+      fields:
+      : {{WebRequestEventDetails/timeStamp}}
+      :: The [=coarsened shared current time=] given
+          |crossOriginIsolatedCapability|.
+
+      : {{WebRequestEventDetails/url}}
+      :: The last element of |request|'s [=request/URL list=].
+
+      : {{WebRequestEventDetails/method}}
+      :: |request|'s [=request/method=].
+
+      : {{WebRequestEventDetails/initiator}}
+      :: The [=serialization of an origin=] given |request|'s [=request/origin=].
+
+         Note: An opaque origin will result in {{WebRequestEventDetails/initiator}}
+         being set to the [=string=] "null".
+
+      : {{WebRequestEventDetails/type}}
+      :: The result of calling [=get a request's ResourceType=] given |request|.
+
+      : {{WebRequestEventDetails/requestId}}
+      :: |request|'s [=request/requestId=].
+
+      : {{WebRequestEventDetails/frameId}}
+      :: -1
+
+      : {{WebRequestEventDetails/parentFrameId}}
+      :: -1
+
+  1. If |environment settings object|'s [=environment settings object/global object=]
+      is a {{Window}} object, then:
+
+      1. Let |window| be |environment settings object|'s
+          [=environment settings object/global object=].
+
+      1. Update the following fields of |details|:
+
+          : {{WebRequestEventDetails/documentId}}
+          :: |environment settings object|'s [=environment/id=].
+
+          : {{WebRequestEventDetails/documentLifecycle}}
+          :: <span class=XXX>It's unclear if {{DocumentLifecycle}} values map
+              to existing concepts.</span>
+
+          : {{WebRequestEventDetails/frameId}}
+          :: The [=navigable/frameId=] of |window|'s [=window/navigable=].
+
+          : {{WebRequestEventDetails/frameType}}
+          :: {{FrameType/"fenced_frame"}} if |window|'s {{Window/fence}} is
+              non-null, {{FrameType/"outermost_frame"}} if |window|'s
+              {{Window/top}} equals |window|, {{FrameType/"sub_frame"}}
+              otherwise.
+
+          1. If |window|'s {{Window/parent}} is not equal to |window|, then
+              update the following values of |details|:
+
+              : {{WebRequestEventDetails/parentDocumentId}}
+              :: The [=environment/id=] of |window|'s {{Window/parent}}'s
+                  [=active document=]'s [=relevant global object=].
+
+              : {{WebRequestEventDetails/parentFrameId}}
+              :: The [=navigable/frameId=] of |window|'s {{Window/parent}}'s
+                  [=window/navigable=].
+
+  1. Return |details|.
+
+</div>
+
+<!--
```

Comment:
Is this comment line intentional?

Reply:
Yes, I had written a few algorithms ahead of time that aren't used yet and didn't want to split that section out into another PR so just commented it out.